### ISSUE TITLE
Subscriptions are not received with pubnub 4.0.7

### ISSUE THREAD

**Issue by grokify** [grokify/ringcentral-sdk-ruby#4]
[PubNub's Ruby SDK](https://github.com/pubnub/ruby) was updated from 3.x to 4.x to use Celluloid in lieu of EventMachine. While the API remains the same and events are executed, something is preventing the subscription callbacks from being executed. This code is in the [`RingCentralSdk::REST::Subscription` class](https://github.com/grokify/ringcentral-sdk-ruby/blob/master/lib/ringcentral_sdk/rest/subscription.rb#L199).

Examined two approaches so far:
1. `Pubnub#subscribe` now returns a `Celluloid::Future` object. Did some initial tests such as saving the `Celluloid::Future` as an object property so it doesn't go out of scope but that didn't solve the issue.
2. Changed the `.subscribe` call from using the `callback:` parameter to the [tutorial style](https://www.pubnub.com/docs/ruby/data-streams-publish-and-subscribe), but that also didn't address the issue:

``` ruby
pubnub.subscribe(
  channel: 'my_channel'
) do |envelope| 
  puts envelope.message
end
```

Will do more testing later.

For now, this can be an issue when using RingCentral 1.3.0 and 1.2.3 which use the latest PubNub, e.g. 4.0.7.

RingCentral 1.3.1 handles this by locking to PubNub 3.8 for now.

**Comment by pubnubcraig** [comment]
v4 SDKs use a single listener for all subscribe/presence and subscribe status messages. You can see example here in the docs
https://www.pubnub.com/docs/ruby/pubnub-ruby-sdk-v4#ruby-sdk-usage

**Comment by grokify** [comment]
@pubnubcraig Thanks! That looks like it will help and I'll check it out.

It may be useful to update the [SDK `README.md`](https://github.com/pubnub/ruby/blob/b01941a793dfb7a03f15802f0f6f07b97abfa528/README.md) and the [Ruby Tutorial](https://www.pubnub.com/docs/ruby/data-streams-publish-and-subscribe). The `README.md` says the public API was unchanged which I took to mean no change in code were necessary.

**Comment by grokify** [comment]
Updated to `4.0.16` in master branch.

**Comment by grokify** [comment]
Resolved in `v2.0.0`
